### DISCUSSION TITLE
Allow users to download tutorial code demo to quickly get their feet wet

### DISCUSSION THREAD

**Discussion by nh916** [django/channels#2032] [Ideas]
When I was following the tutorial, I kept thinking "I wish this tutorial code already existed somewhere so I can mess with it quickly and see if it is what I need or not before taking so long to follow the documentation." and "I wish the documentation example code existed somewhere so I can just download it and start experimenting with it"

I ended up writing copying the exact code from the tutorial and put it into a repository a while ago, I am wondering if the maintainers thing it would be a good idea if I clean up the code a bit, add some comments, docstrings, type hints, and try to merge it into this codes documentation, so new users can easily and quickly get their feet wet with the demo.

My current repository with Django Channels Documentation Tutorial:
https://github.com/nh916/Django-Channels-Documentation-tutorial

**Comment by carltongibson** [comment]
Hi @nh916 — thanks for the suggestion here. 

I also have [my own version on the channels tutorial](https://github.com/carltongibson/notes/tree/main/examples/channels_tutorial) — but I have to say it's not something that I'd want to make part of the official repo. 

* My experience of these things is that they aren't kept up to date, and become just another maintenance headache. 
* But also, perhaps more, I don't think it's good for users to have to tutorial too easily downloadable. The analogy is with working out: you can't just read a book about it — you actually have to work out. Code exercises are like that. You should always actually do them. It's how you built the muscles. 

I hope that makes sense.
